bundles/framework/announcements/view/AnnouncementsHandler.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { StateHandler, controllerMixin, Messaging } from 'oskari-ui/util';
+import { Message } from 'oskari-ui';
+
+// Handler for announcements. Handles state and service calls.
+
+const getMessage = (key, args) => <Message messageKey={key} messageArgs={args} bundleKey='announcements' />;
+
+class ViewHandler extends StateHandler {
+    constructor (instance) {
+        super();
+        this.instance = instance;
+        this.sandbox = instance.getSandbox();
+        this.locale = instance.getLocalization();
+        this.state = {
+            panels: [],
+            modals: [],
+            updated: false,
+            announcements: [],
+            checked: false
+        };
+    }
+
+    getAnnouncements (callback) {
+        if (typeof callback !== 'function') {
+            return;
+        }
+
+        jQuery.ajax({
+            type: 'GET',
+            dataType: 'json',
+            url: Oskari.urls.getRoute('GetAnnouncements'),
+            success: function (pResp) {
+                callback(pResp);
+            },
+            error: function (jqXHR, textStatus) {
+                Messaging.error(getMessage('messages.getFailed'));
+            }
+        });
+    }
+
+    updatePanelsModals (panels, modals) {
+        this.updateState({
+            panels: panels,
+            modals: modals,
+            updated: true
+        });
+    }
+
+    handleOk (index, title, content, checked) {
+        if (checked) {
+            localStorage.setItem(title, content);
```

Comment:
This feels very fragile approach in a way that if you have an announcement with title "Welcome" there and some other functionality that would like to store something with key "Welcome" to the localstorage these will instantly conflict. Also the content is never used? Why not just use a boolean instead so it doesn't use space on the users device as much?

Refactor this so that instead of "title" as key you use something like "oskari-announcements" and as a value store an object with title and boolean as value. Or better yet if you have an id you can use that instead of the title so different language versions don't affect each other. Instead of an object as localstorage value you could use an array with "what ever is on the array has been shown" whether it's the title or id. 

Now if the admin changes the title of the announcement it is shown again. Or if the user changes the language I suppose the announcement title changes and is shown again. Have you considered using something like the announcement add/modify/save date to detect if the announcement should be shown? Then showing the announcement again could be triggered if the admin saves any modifications to the announcement (changing the content, not the title like it is now).

Reply:
refactored